woocommerce/woocommerce-android
# [Woo POS][Non Simple product types] Fetch variations list

Closes: #12844 


## NOTE: I've extracted the UI in the products screen into separate file which is the cause for many line changes.

### Description

This PR fetches variations list and displays on the variations screen.

It makes use of existing `VariationsListHandler` class for fetching variations.

### Testing information


**Testing Variations list**
1. Navigate to POS mode (more menu -> POS)
2. Click on any variable product
3. Ensure you see variations displayed

**Testing Variations pagination**
1. Navigate to POS mode (more menu -> POS)
2. Click on any variable product that has more than 10 variations
3. Ensure you see variations displayed and it paginates as expected

**Testing Variations pagination error**
1. Navigate to POS mode (more menu -> POS)
2. Click on any variable product that has more than 10 variations
3. Ensure you see variations displayed
4. Turn on airplane mode
5. Scroll down to end of the list
6. Ensure you see snackbar message that says failed to fetch variations.

**Testing Variations fetch error**
1. Navigate to POS mode (more menu -> POS)
2. Turn on airplane mode
3. Click on any variable product
4. Ensure you see appropriate error screen


### The tests that have been performed

Tested all the above scenario listed in both light and dark modes on emulator.

### Images/gif



## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
